survival_sponsorship_compassion/views/sponsorship_contract_view.xml

```diff
@@ -0,0 +1,17 @@
+<odoo>
+    <record id="survival_sponsorship_product_filter_tree_view" model="ir.ui.view">
+        <field name="name">survival.sponsorship.product.filter.tree</field>
+        <field name="model">recurring.contract.line</field>
+        <field name="inherit_id" ref="sponsorship_compassion.view_compassion_sponsorship_line_tree"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='product_id']" position="attributes">
+                <attribute name="domain">
+                    [('survival_sponsorship_sale', '=', active_ids[0].type == 'CSP')]
```

Comment:
@ecino  do you have any idea why my domain doesn't work ? I tried many things but nothing is working right.
The goal is to only show the Survival product in case the type of the sponsorship is survival.